sql/src/main/java/io/crate/metadata/Schemas.java

```diff
@@ -122,36 +125,77 @@ public TableInfo resolveTableInfo(QualifiedName ident, Operation operation, User
+                } else {
+                    throw RelationUnknown.of(tableName, getSimilarTables(user, tableName, currentSchema.getTables()));
+                }
+            }
         } else {
             schemaInfo = schemas.get(identSchema);
             if (schemaInfo == null) {
                 throw SchemaUnknownException.of(identSchema, getSimilarSchemas(user, identSchema));
             } else {
                 tableInfo = schemaInfo.getTableInfo(tableName);
+                if (tableInfo == null) {
+                    throw RelationUnknown.of(ident.toString(), getSimilarTables(user, tableName, schemaInfo.getTables()));
+                }
             }
         }
-        if (tableInfo == null) {
-            throw new RelationUnknown(ident.toString());
-        }
         Operation.blockedRaiseException(tableInfo, operation);
         return tableInfo;
     }
 
+    private static List<String> getSimilarTables(User user, String tableName, Iterable<TableInfo> tables) {
+        LevenshteinDistance levenshteinDistance = new LevenshteinDistance();
+        ArrayList<Candidate> candidates = new ArrayList<>();
+        for (TableInfo table : tables) {
+            if (user.hasAnyPrivilege(Privilege.Clazz.TABLE, table.ident().fqn())) {
+                String candidate = table.ident().name();
+                float score = levenshteinDistance.getDistance(tableName.toLowerCase(Locale.ENGLISH), candidate.toLowerCase(Locale.ENGLISH));
+                if (score > 0.7f) {
+                    candidates.add(new Candidate(score, candidate));
+                }
+            }
+        }
+        candidates.sort(Comparator.comparing((Candidate x) -> x.score).reversed());
+        return candidates.stream()
+            .limit(5)
+            .map(x -> x.name)
+            .collect(Collectors.toList());
+    }
+
     private List<String> getSimilarSchemas(User user, String schema) {
         LevenshteinDistance levenshteinDistance = new LevenshteinDistance();
-        ArrayList<String> candidates = new ArrayList<>();
+        ArrayList<Candidate> candidates = new ArrayList<>();
         for (String availableSchema : schemas.keySet()) {
             if (user.hasAnyPrivilege(Privilege.Clazz.SCHEMA, availableSchema)) {
-                if (levenshteinDistance.getDistance(schema, availableSchema) > 0.7f) {
-                    candidates.add(availableSchema);
-                } else if (schema.equalsIgnoreCase(availableSchema)) {
-                    candidates.add(availableSchema);
+                float score = levenshteinDistance.getDistance(schema.toLowerCase(Locale.ENGLISH), availableSchema.toLowerCase(Locale.ENGLISH));
+                if (score > 0.7f) {
+                    candidates.add(new Candidate(score, availableSchema));
                 }
             }
         }
-        return candidates;
+        candidates.sort(Comparator.comparing((Candidate x) -> x.score).reversed());
+        return candidates.stream()
```

Comment:
we could have use a priority queue as well, but it is probably not that critical 👍